hend.py

```diff
@@ -0,0 +1,62 @@
+    '''
+    @author: dhend
+    This addresses the following variables: exterior1st, exterior2nd, roofmatl, lotarea, lotfrontage, masvnrarea, and masvnrtype.
+    Most of the data remains unaffected.
+
+        Parameters
+    ----------
+    df : pd.DataFrame
+        This should be the entire Housing Prices dataframe.
+    Returns
+    -------
+    pd.DataFrame
+        This dataframe will include ONLY the engineered features
+        listed above.
+    '''
+
+import pandas as pd
+def feature_extract(df):
+
+    #Exterior2nd was nearly identical to Exterior1st, so I dropped it.
+    #df.drop(['Exterior2nd'], axis=1, inplace = True)
+
+    #Imputing missing numerical data with the mean and categorical data with the mode.
+    df['LotFrontage'].fillna(df['LotFrontage'].mean(), inplace=True)
+    df['MasVnrArea'].fillna(df['MasVnrArea'].mode(), inplace=True)
+
+    #Determing interquartile range 
+    Q1 = df['LotFrontage'].quantile(.25)
+    Q3 = df['LotFrontage'].quantile(.75)
+    IQR = Q3 - Q1
+
+    #I replaced outliers with the mean.
+    for entry in df['LotFrontage']:
+        if entry > (Q3 + 1.5 * IQR):
+            df['LotFrontage'].replace(entry, df['LotFrontage'].mean(), inplace = True)
+
+    Q1 = df['MasVnrArea'].quantile(.25)
+    Q3 = df['MasVnrArea'].quantile(.75)
+    IQR = Q3 - Q1
+
+    for entry in df['MasVnrArea']:
+    if entry > (Q3 + 1.5 * IQR):
+        df['MasVnrArea'].replace(entry, df['MasVnrArea'].mean(), inplace = True)
+
+    #Dummy variables are given for categorical variables.
+    pd.get_dummies(df['Exterior1st'])
+    pd.get_dummies(df['RoofMatl'])
+    pd.get_dummies(df['MasVnrType'])
+

```

Comment:
get_dummies doesn't put them into the dataframe. It returns them. You'll need to capture them in a variable and add them back to your frame.